File: Producto/frontend/src/remote/DTO/NVTrazabilidadResumenDTO.js
/**
 * Resumen de una Nota de Venta con sus Ordenes de Produccion asociadas,
 * usado por el listado general de trazabilidad (TrazabilidadNV).
 * Espejo de NVTrazabilidadResumenDTO.java del backend.
 */
export class NVTrazabilidadResumenDTO {
    constructor(data = {}) {
        this.idNV = data.idNV ?? null;
        this.numeroNV = data.numeroNV ?? '';
        this.clienteId = data.clienteId ?? null;
        this.clienteNombre = data.clienteNombre ?? '';
        this.estado = data.estado ?? null;
        this.fechaEmision = data.fechaEmision ?? null;
        this.motivoRechazo = data.motivoRechazo ?? null;
        this.fechaRechazo = data.fechaRechazo ?? null;

        this.evaluacionNegocioId = data.evaluacionNegocioId ?? null;
        this.numeroEVN = data.numeroEVN ?? null;
        this.estadoEVN = data.estadoEVN ?? null;
        this.motivoRechazoEVN = data.motivoRechazoEVN ?? null;
        this.fechaRechazoEVN = data.fechaRechazoEVN ?? null;

        this.ordenesProduccion = Array.isArray(data.ordenesProduccion)
            ? data.ordenesProduccion.map(op => ({
                idOP: op.idOP ?? null,
                numeroOP: op.numeroOP ?? '',
                estado: op.estado ?? null,
                fechaEntregaProgramada: op.fechaEntregaProgramada ?? null,
                hojaCompra: op.hojaCompra ? {
                    idHC: op.hojaCompra.idHC ?? null,
                    numeroHC: op.hojaCompra.numeroHC ?? '',
                    estado: op.hojaCompra.estado ?? null,
                } : null,
                ordenesCompra: Array.isArray(op.ordenesCompra)
                    ? op.ordenesCompra.map(oc => ({
                        idOC: oc.idOC ?? null,
                        numeroOC: oc.numeroOC ?? '',
                        estado: oc.estado ?? null,
                        motivoRechazo: oc.motivoRechazo ?? null,
                        fechaRechazo: oc.fechaRechazo ?? null,
                    }))
                    : [],
            }))
            : [];
    }

    static listFromResponse(response) {
        if (!response || !Array.isArray(response.data)) return [];
        return response.data.map(item => new NVTrazabilidadResumenDTO(item));
    }
}

File: Producto/backend/src/main/java/backend/com/comercial/application/dto/NVTrazabilidadResumenDTO.java
package backend.com.comercial.application.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class NVTrazabilidadResumenDTO {
    private Long idNV;
    private String numeroNV;
    private Long clienteId;
    private String clienteNombre;
    private String estado;
    private LocalDate fechaEmision;
    private String motivoRechazo;
    private LocalDateTime fechaRechazo;

    private Long evaluacionNegocioId;
    private String numeroEVN;
    private String estadoEVN;
    private String motivoRechazoEVN;
    private LocalDateTime fechaRechazoEVN;

    private List<OPResumenDTO> ordenesProduccion;

    @Data
    @Builder
    public static class OPResumenDTO {
        private Long idOP;
        private String numeroOP;
        private String estado;
        private LocalDate fechaEntregaProgramada;
        private HCResumenDTO hojaCompra;
        private List<OCResumenDTO> ordenesCompra;
    }

    @Data
    @Builder
    public static class HCResumenDTO {
        private Long idHC;
        private String numeroHC;
        private String estado;
    }

    @Data
    @Builder
    public static class OCResumenDTO {
        private Long idOC;
        private String numeroOC;
        private String estado;
        private String motivoRechazo;
        private LocalDateTime fechaRechazo;
    }
}
